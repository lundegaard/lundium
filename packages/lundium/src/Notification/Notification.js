import React from 'react';
import PropTypes from 'prop-types';
import { cx } from 'ramda-extension';
import { T, always, cond, equals, identity } from 'ramda';

import Grid from '../Grid';
import Box from '../Box';
import Icon from '../Icon';
import Button from '../Button';

const getIcon = cond([
	[equals('danger'), always('status-rejected')],
	[equals('warning'), always('error')],
	[T, identity],
]);

// TODO Mobile should not have A close button,
// desired functionality is to hide notification with swipe up
// (just like native notifications)
const Notification = ({
	children,
	className,
	type = 'success',
	onClose,
	id,
	...otherProps
}) => (
	<Grid flex className={cx('notification', type, className)} {...otherProps}>
		<Box className="notification__icon" mx={3}>
			<Icon type={getIcon(type)} size={22} color={type} />
		</Box>
		<Box className="notification__text">{children}</Box>
		<Box mx={3}>
			<Button
				className="py-0"
				kind="blank"
				size="xs"
				onClick={() => onClose(id)}
			>
				<Icon type="cancel" size={16} />
			</Button>
		</Box>
	</Grid>
);

Notification.propTypes = {
	children: PropTypes.node,
	className: PropTypes.string,
	icon: PropTypes.string,
	id: PropTypes.string,
	onClose: PropTypes.func,
	type: PropTypes.string,
};

export const NotificationTypes = [
	'danger',
	'warning',
	'info',
	'success',
	'primary',
	'secondary',
];

export default Notification;
