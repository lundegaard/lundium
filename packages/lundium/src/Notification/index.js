export { default } from './Notification';
export { NotificationTypes } from './Notification';
