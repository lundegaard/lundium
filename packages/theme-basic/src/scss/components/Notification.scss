@import '../core';

/*
 -  Notification
 -  (<div class="notification $theme-color" />)
 */


.notification {
	width: 100%;
	max-width: $notification-width;
	padding-block: $spacer $spacer;
	background-color: $notification-bg;
	border-radius: $border-radius;
	border-inline-start: 10px solid $theme-gray-300;
	box-shadow: $notification-box-shadow;
	margin-block-start: $spacer;

	@each $color-name, $color in $theme-colors {
		&.#{$color-name} {
			border-inline-start-color: $color;
		}
	}

	&__icon {
		margin-block-start: 2px;
	}

	&__text {
		line-height: ( 28 / 14 );
		flex-grow: 1;

		p {
			margin-block-end: 0;
		}
	}
}