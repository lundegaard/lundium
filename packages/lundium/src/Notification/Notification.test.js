import React from 'react';
import { shallow } from 'enzyme';

import Notification from './Notification';

describe('<Notification />', () => {
	const requiredProps = {};

	it('matches snapshot when shallowly rendered', () => {
		const wrapper = shallow(<Notification {...requiredProps} />);
		expect(wrapper).toMatchSnapshot();
	});
});
