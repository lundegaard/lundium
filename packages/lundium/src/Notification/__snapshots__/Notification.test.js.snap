// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notification /> matches snapshot when shallowly rendered 1`] = `
<forwardRef(Grid)
  className="notification success"
  flex={true}
>
  <forwardRef(Box)
    className="notification__icon"
    mx={3}
  >
    <Icon
      color="success"
      size={22}
      type="success"
    />
  </forwardRef(Box)>
  <forwardRef(Box)
    className="notification__text"
  />
  <forwardRef(Box)
    mx={3}
  >
    <ForwardRef
      className="py-0"
      kind="blank"
      onClick={[Function]}
      size="xs"
    >
      <Icon
        size={16}
        type="cancel"
      />
    </ForwardRef>
  </forwardRef(Box)>
</forwardRef(Grid)>
`;
